### ISSUE TITLE
Enable sorting / reverse order on time entries and other lists

### ISSUE THREAD

**Issue by joleaf** [te-online/timemanager#25]
Current State:

The time entries are sorted by the timestamp:
![image](https://user-images.githubusercontent.com/71190322/107346950-98d30580-6ac5-11eb-9d7a-ae59923589a0.png)

(Furthermore, the german date format is wrong :( )

Expected State:

The time entries are sorted by the timestamp but in descending order.

**Comment by te-online** [comment]
@joleaf I can see your point :-) Let's break this up, though.

> The time entries are sorted by the timestamp:

I'm not convinced that reversing the entries is the solution here. We can anticipate that users might be used to this sorting, so we shouldn't mess with the default in my opinion. It's reasonable to sort from oldest to newest from top to bottom in my view. 

Having said that, I'd be open to adding a sort toggle of sorts. Any ideas on that?

> (Furthermore, the german date format is wrong :( )

This is a bug that I will try to address 👍

**Comment by joleaf** [comment]
@te-online yes, you are totally right, a toggle button might be the solution :)

**Comment by 31SFX4** [comment]
I also need it usually with the newest entries at the top, so a toggle button would be great (especially when a task has very many entries so you have to scroll very far to see the newest). 

By the way, what do the checkboxes in the list actually do?

**Comment by te-online** [comment]
> By the way, what do the checkboxes in the list actually do?

The checkmarks are a "done"/"waiting" indication for your purpose of choice. Many people use it for invoicing – unchecked = unpaid, checked = paid.

**Comment by raph-topo** [comment]
May I suggest that the toggle to switch ascending & descending order may be added:
- on all lists: Clients, Projects, Tasks, Time Entries & Reports
- each for all properties: time created (useful mostly for tasks & entries), start time/end time (tasks, entries, reports) name (clients, projects, tasks, reports), length (projects, tasks, entries, reports).

Suggested new issue title: "Add various sorting options to all lists"

**Comment by te-online** [comment]
Hi @raph-topo – thanks for your input – I appreciate it a lot 😊 

I like the idea to put sorting on all lists, that's very good!

In the spirit of keeping this app as simple as possible, I'd refrain from adding too many options. 🙏 💫 

I'd suggest we start out with, for **time entries**
- reversing the default sorting to always be "end date – newest to oldest", because I think that's what most people would prefer (or rather, what I'd personally prefer)
- allowing to manually reverse the sorting so oldest are first (the old behavior)

Clients are already sorted alphabetically, if I remember correctly.

Projects and tasks are sorted by "created at", if I remember correctly.

- For all three lists, I'd suggest to keep the existing sorting and adding the ability to reverse the existing sorting
- For projects and tasks we could add sorting alphabetically (and reverse) as an option
- For clients I'd find a "last activity" sorting (and reverse) useful

What do you think?

**Comment by raph-topo** [comment]
Hey,
I was thinking it could be as simple as Nextcloud's default interface, like Files list:
![image](https://github.com/te-online/timemanager/assets/53143154/952e2404-fc4e-46a4-8bf0-5e0d4c9ac85b)

**Comment by te-online** [comment]
That may be a little difficult, because I'm not showing the data in columns like the Files app 😊 

But it's not a bad place to start to look for consistency with the rest of Nextcloud 👍

**Comment by te-online** [comment]
In the next version, time entry sorting will be reversed. For now, there's no way to revert it back to the old behavior, but I may add that in the future. Especially, if there is demand for it.
